internal/app/proxyd/main.go

```diff
@@ -28,16 +34,51 @@ func main() {
 		log.Fatalf("open user data: %v", err)
 	}
 
-	r, err := frontend.NewReverseProxy(data.Services.Trustd.Endpoints)
+	bootstrapCtx, bootstrapCancel := context.WithCancel(context.Background())
+	r, err := frontend.NewReverseProxy(data.Services.Trustd.Endpoints, bootstrapCancel)
 	if err != nil {
 		log.Fatalf("failed to initialize the reverse proxy: %v", err)
 	}
 
+	// Start up with initial bootstrap config
+	go r.Bootstrap(bootstrapCtx)
+
 	// nolint: errcheck
-	go r.Listen(":443")
+	go func() {
+		kubeconfig := "/etc/kubernetes/admin.conf"
+		if err = conditions.WaitForFilesToExist(kubeconfig).Wait(context.TODO()); err != nil {
```

Comment:
`s/context.TODO/context.Background/`. `TODO` is more of a placeholder when you don't know which context to use.